src/main/java/session/MenuSession.java

```diff
@@ -127,8 +130,26 @@ private void initializeBreadcrumbs() {
         this.breadcrumbs.add(SessionUtils.getAppTitle());
     }
     
-    public void updateApp(String updateMode) {
-        this.engine.attemptAppUpdate(updateMode);
+    public NotificationMessage updateApp(String updateMode) {
+        try {
+            if (this.engine.attemptAppUpdate(updateMode)) {
+                return new NotificationMessage("Application updated successfully.", false);
+            } else {
+                return new NotificationMessage("Application up to date.", false);
+            }
+        } catch (UnresolvedResourceException e) {
+            String message = "Update Failed! Couldn't find or install one of the remote resources";
+            log.error(message, e);
+            return new NotificationMessage(message, true);
+        } catch (UnfullfilledRequirementsException e) {
+            String message = "Update Failed! This CLI host is incompatible with the app";
+            log.error(message, e);
+            return new NotificationMessage(message, true);
+        } catch (Exception e) {
+            String message = "Update Failed! There is a problem with one of the resources";
```

Comment:
Should this message be more generic, given that we really have no idea where this Exception came from if we get to this point?

Reply:
Good point; I copied this code and it looks like this catch was needlessly generic, the only typed exception remaining was `InstallCancelledException` which I'm now catching explicitly